From oneukum@suse.de Tue Feb 27 01:25:15 2007
From: Oliver Neukum <oneukum@suse.de>
Date: Tue, 27 Feb 2007 10:25:00 +0100
Subject: USB: ratelimit debounce error messages
To: gregkh@suse.de, linux-usb-devel@lists.sourceforge.net
Message-ID: <200702271025.00634.oneukum@suse.de>
Content-Disposition: inline

flaky hardware can cause a lot of debounce failed messages. To limit
the performance impact, a ratelimit should be used.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/hub.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/core/hub.c
+++ b/drivers/usb/core/hub.c
@@ -2443,7 +2443,7 @@ static void hub_port_connect_change(stru
 
 	if (portchange & USB_PORT_STAT_C_CONNECTION) {
 		status = hub_port_debounce(hub, port1);
-		if (status < 0) {
+		if (status < 0 && printk_ratelimit()) {
 			dev_err (hub_dev,
 				"connect-debounce failed, port %d disabled\n",
 				port1);
